service/create/service.go

```diff
@@ -492,10 +519,21 @@ func (s *Service) runMachine(input runMachineInput) error {
 			},
 		},
 	}); err != nil {
-		return microerror.MaskAny(err)
+		return nil, microerror.MaskAny(err)
 	}
 
 	s.logger.Log("info", fmt.Sprintf("instance '%s' tagged", input.name))
+	instance := reservation.Instances[0]
 
-	return nil
+	return instance, nil
```

Comment:
> When we create resources the interface of the creating function should return an error and an identifier which can be used to fetch information about the created resources using another function which job is only to fetch information.

So IIUC, making the function return `instanceID` and then, whenever we need to get information about the instance, call `DescribeInstances()` with `instanceID` is what we should do. Right?

This is what the PR did at the beginning, maybe the missing piece of information was that with `instanceID` you can get all kinds of information using `DescribeInstances()` in case we need more info in the future.